File: SWIG/Examples/test-suite/python/template_typedef_cplx2_runme.py
import string
from template_typedef_cplx2 import *

#
# double case
#

try:
  d = make_Identity_double()
  a = d.this
except:
  print d, "is not an instance"
  raise RuntimeError

s = string.split('%s' % d)[1]
if string.find(s, 'ArithUnaryFunction') == -1:
  print d, "is not an ArithUnaryFunction"
  raise RuntimeError

try:
  e = make_Multiplies_double_double_double_double(d, d)
  a = e.this
except:
  print e, "is not an instance"
  raise RuntimeError

s = string.split('%s' % e)[1]
if string.find(s, 'ArithUnaryFunction') == -1:
  print e, "is not an ArithUnaryFunction"
  raise RuntimeError


#
# complex case
#

try:
  c = make_Identity_complex()
  a = c.this
except:
  print c, "is not an instance"
  raise RuntimeError

s = string.split('%s' % c)[1]
if string.find(s, 'ArithUnaryFunction') == -1:
  print c, "is not an ArithUnaryFunction"
  raise RuntimeError

try:
  f = make_Multiplies_complex_complex_complex_complex(c, c)
  a = f.this
except:
  print f, "is not an instance"
  raise RuntimeError

s = string.split('%s' % f)[1]
if string.find(s, 'ArithUnaryFunction') == -1:
  print f, "is not an ArithUnaryFunction"
  raise RuntimeError

#
# Mix case
#

try:
  g = make_Multiplies_double_double_complex_complex(d, c)
  a = g.this
except:
  print g, "is not an instance"
  raise RuntimeError

s = string.split('%s' % g)[1]
if string.find(s, 'ArithUnaryFunction') == -1:
  print g, "is not an ArithUnaryFunction"
  raise RuntimeError


try:
  h = make_Multiplies_complex_complex_double_double(c, d)
  a = h.this
except:
  print h, "is not an instance"
  raise RuntimeError

s = string.split('%s' % g)[1]
if string.find(s, 'ArithUnaryFunction') == -1:
  print g, "is not an ArithUnaryFunction"
  raise RuntimeError

try:
  a = g.get_value()
except:
  print g, "has not get_value() method"
  raise RuntimeError

File: Examples/test-suite/lua/template_typedef_cplx2_runme.lua
local v=require 'template_typedef_cplx2'

-- double case
local d = v.make_Identity_double()
assert(swig_type(d) == "vfncs::ArithUnaryFunction< vfncs::arith_traits< double,double >::argument_type,vfncs::arith_traits< double,double >::result_type > *|vfncs::unary_func_traits< double,double >::base *|vfncs::ArithUnaryFunction< double,double > *")
local e = v.make_Multiplies_double_double_double_double(d, d)
assert(swig_type(e) == "vfncs::ArithUnaryFunction< vfncs::arith_traits< double,double >::argument_type,vfncs::arith_traits< double,double >::result_type > *|vfncs::unary_func_traits< double,double >::base *|vfncs::ArithUnaryFunction< double,double > *")

-- complex case
local c = v.make_Identity_complex()
assert(swig_type(c) == "vfncs::ArithUnaryFunction< Complex,Complex > *|vfncs::ArithUnaryFunction< vfncs::arith_traits< std::complex< double >,std::complex< double > >::argument_type,vfncs::arith_traits< std::complex< double >,std::complex< double > >::result_type > *|vfncs::unary_func_traits< std::complex< double >,std::complex< double > >::base *|vfncs::ArithUnaryFunction< std::complex< double >,std::complex< double > > *")
local f = v.make_Multiplies_complex_complex_complex_complex(c, c)
assert(swig_type(f) == "vfncs::ArithUnaryFunction< Complex,Complex > *|vfncs::ArithUnaryFunction< vfncs::arith_traits< std::complex< double >,std::complex< double > >::argument_type,vfncs::arith_traits< std::complex< double >,std::complex< double > >::result_type > *|vfncs::unary_func_traits< std::complex< double >,std::complex< double > >::base *|vfncs::ArithUnaryFunction< std::complex< double >,std::complex< double > > *")

-- Mix case
local g = v.make_Multiplies_double_double_complex_complex(d, c)
assert(swig_type(g) == "vfncs::ArithUnaryFunction< double,Complex > *|vfncs::ArithUnaryFunction< vfncs::arith_traits< double,std::complex< double > >::argument_type,vfncs::arith_traits< double,std::complex< double > >::result_type > *|vfncs::ArithUnaryFunction< vfncs::arith_traits< std::complex< double >,double >::argument_type,vfncs::arith_traits< std::complex< double >,double >::result_type > *|vfncs::ArithUnaryFunction< double,std::complex< double > > *")

-- This should raise NoMethodError if it fails
assert(g:get_value() == 1)
local h = v.make_Multiplies_complex_complex_double_double(c, d)
assert(swig_type(h) == "vfncs::ArithUnaryFunction< double,Complex > *|vfncs::ArithUnaryFunction< vfncs::arith_traits< double,std::complex< double > >::argument_type,vfncs::arith_traits< double,std::complex< double > >::result_type > *|vfncs::ArithUnaryFunction< vfncs::arith_traits< std::complex< double >,double >::argument_type,vfncs::arith_traits< std::complex< double >,double >::result_type > *|vfncs::ArithUnaryFunction< double,std::complex< double > > *")
